### PR TITLE
Clean up GC_Finalizable*Buffer classes code

### PR DESCRIPTION
**Pull request by dmitripivkine**
Use correct types, fix formatting

### CODE REVIEW

**Comment by dmitripivkine** [comment]
fixes https://github.com/eclipse-openj9/openj9/issues/20458

**Comment by amicic** [comment]
do we care about spaces after the comma (I think most of the time we have them)?

```
	GC_FinalizableReferenceBuffer(MM_GCExtensions *extensions) :
		_head(NULL)
		,_tail(NULL)
		,_count(0)
		,_extensions(extensions)
```

**Comment by dmitripivkine** [comment]
yes, good point. I am going to fix it.

**Review by amicic — APPROVED** [review]

**Comment by amicic** [comment]
jenkins compile aix jdk17

**Review by LinHu2016 — APPROVED** [review]

**Review by LinHu2016 — APPROVED** [review]
LGTM

File: runtime/gc_base/FinalizableClassLoaderBuffer.hpp [hunk 1]
```diff
@@ -88,9 +88,9 @@ class GC_FinalizableClassLoaderBuffer
 	 */
 	GC_FinalizableClassLoaderBuffer(MM_GCExtensions *extensions) :
 		_head(NULL)
-		,_tail(NULL)
-		,_count(0)
-		,_extensions(extensions)
+		, _tail(NULL)
+		, _count(0)
+		, _extensions(extensions)
 	{}
```
Comment (keithc-ca) [inline runtime/gc_base/FinalizableClassLoaderBuffer.hpp:94]: I think this pattern is an improvement: it reads better, additions or removals (at the end) don't change any other lines, but I suggest the first field should line up with the rest by moving the colon:
```c
	GC_FinalizableClassLoaderBuffer(MM_GCExtensions *extensions)
		: _head(NULL)
		, _tail(NULL)
		, _count(0)
		, _extensions(extensions)
	{}
```

**Pull request merged**